New vehicle data HandsOffSteering (#839)
smartdevicelink/sdl_evolution: proposals/NNNN-New-vehicle-data-HandsOffSteering.md

# New vehicle data HandsOffSteering

* Proposal: [SDL-NNNN](NNNN-New-vehicle-data-HandsOffSteering.md)
* Author: [Ankur Tiwari](https://github.com/atiwari9)
* Status: **Awaiting review**
* Impacted Platforms: [Core | HMI | Policy Server | SHAID | iOS | Java Suite | RPC]

## Introduction

> This proposal is to add new vehicle data `HandsOffSteering` to know if driver has hands on steering or not.

Reviewer: ```suggestion
This proposal is to add new vehicle data `HandsOffSteering` to know whether or not a driver has his or her hands on the steering wheel.
```

Reviewer: Please also make this change in the PR description.

Author: Done

## Motivation

> In order to partner with more diverse app partners, we need to provide additional sets of vehicle data items through SDL. This goes in line with commitment to enhance SDL with even richer vehicle data content. Information about whether driver has hands on steering or not is useful for apps that compute driver score and provide insurance.

Reviewer: ```suggestion
In order to partner with more diverse app partners, we need to provide additional sets of vehicle data items through SDL. This goes in line with commitment to enhance SDL with even richer vehicle data content. Information about whether a driver has hands on the steering wheel or not is useful for apps that compute driver score and provide insurance.
```

Author: Done

```xml	
<element name="VEHICLEDATA_HANDSOFFSTEERING"/>
```

#### Add the following parameter to these function requests:
* `SubscribeVehicleData`
* `UnsubscribeVehicleData`
* `GetVehicleData`

```xml	
<param name="handsOffSteering" type="Boolean" mandatory="false">
	<description>To indicate whether driver hands are off the steering wheel</description>
</param>
```

#### Add the following parameter to these function responses:
* `SubscribeVehicleData`
* `UnsubscribeVehicleData`

```xml	
<param name="handsOffSteering" type="Common.VehicleDataResult" mandatory="false">
	<description>To indicate whether driver hands are off the steering wheel</description>
</param>
```

#### Add the following parameter to these function responses:
* `GetVehicleData`
* `OnVehicleData`

```xml	
<param name="handsOffSteering" type="Boolean" mandatory="false">
	<description>To indicate whether driver hands are off the steering wheel</description>
</param>
```

## Potential downsides

Author is not aware of any downsides to proposed solution. This proposal just enhances the SDL content.

## Impact on existing code

* SDL Core needs to be updated as per new API.

> * iOS/Java Suite needs to be updated to support getters/setters as per new API.

Reviewer: ```suggestion
* iOS/Java Suite need to be updated to support getters/setters as per new API.
```

Author: Done
